packages/twenty-server/src/engine/core-modules/auth/auth.resolver.ts

```diff
@@ -262,6 +274,32 @@ export class AuthResolver {
     };
   }
 
+  @UseGuards(CaptchaGuard)
+  @Mutation(() => SignUpOutput)
+  async createUserAndWorkspace(
+    @Args() createUserAndWorkspaceInput: CreateUserAndWorkspaceInput,
+  ): Promise<SignUpOutput> {
+    const { user, workspace } = await this.signInUpService.signUpOnNewWorkspace(
```

Comment:
are we allowing to sign up any existing user email to a new workspace here? This seems wrong to, the user / user emails should be fetched from Auth. We let to much be done without being authenticated

Reply:
This feels dangerous to me looks like a pattern that could lead to security breaches

Reply:
Not sure if there is any other product offering this

Reply:
If we want an authenticated user we will need to display only workspaces compatible with the auth method.

It's doable but the experience will be strange.

Reply:
I was also expecting to see this post-auth and was surprised when trying the feature. Then I thought why not 😏. 
But I didn't think of your point Charles, you could trap users into a fake lookalike workspace.

@AMoreaux What about requiring pwd, displaying all workspaces, but it asks you to login again if you pick a workspace with the wrong auth method?